client/src/context/AuthContext.js

```diff
@@ -0,0 +1,95 @@
+import React, { createContext, useState } from "react";
+
+const AuthContext = createContext();
+
+function AuthContextProvider({ children }) {
+  const [auth, setAuth] = useState(false);
```

Comment:
If I logged in and refresh the page, I will be still redirected to /login, but expected to /create-game. This is because auth state initially always false. I think you should set initial state of auth depending on whether you have a cookie or not.